### PR TITLE
Dockerize prosody

### PR DESCRIPTION
**Pull request by lunika**
## Purpose

We want to dockerize Prosody XMPP server. We need some additional
modules in this image: mod_token and mod_token_affiliation.

## Proposal

- [x] Create a Dockerfile installing prosody
- [x] Install additional modules: mod_token and mod_token_affiliation

### CODE REVIEW

File: files/etc/prosody/prosody.cfg.lua [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+-- use Prosody's configured data storage to store the authentication data.
+
+authentication = "internal_hashed"
+
+-- Select the storage backend to use. By default Prosody uses flat files
+-- in its configured data directory, but it also supports more backends
+-- through modules. An "sql" backend is included by default, but requires
+-- additional dependencies. See https://prosody.im/doc/storage for more info.
+
+--storage = "sql" -- Default is "internal"
```
Comment (madmatah) [inline files/etc/prosody/prosody.cfg.lua:160]: I'm wondering about authentication with this image.
The default internal storage used by prosody seems to be stored in `/var/lib/prosody`.

Should this directory be defined as a volume in the Dockerfile, to have persistence ?
Or maybe you want to create the users in the entrypoint based on environment variables ?

WDYT?

File: files/etc/prosody/prosody.cfg.lua [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+certificates = "certs"
+
+-- HTTPS currently only supports a single certificate, specify it here:
+--https_certificate = "certs/localhost.crt"
+
+----------- Virtual hosts -----------
+-- You need to add a VirtualHost entry for each domain you wish Prosody to serve.
+-- Settings under each VirtualHost entry apply *only* to that host.
+
+VirtualHost "localhost"
```
Comment (madmatah) [inline files/etc/prosody/prosody.cfg.lua:206]: What is the purpose of this VirtualHost ?

File: files/etc/prosody/prosody.cfg.lua (hunk 1 above)
Comment (lunika) [reply files/etc/prosody/prosody.cfg.lua:160]: Indeed I should add this directory in the `VOLUME`.

File: files/etc/prosody/prosody.cfg.lua (hunk 2 above)
Comment (lunika) [reply files/etc/prosody/prosody.cfg.lua:206]: To be here as an example, nothing more.

Comment (madmatah) [reply files/etc/prosody/prosody.cfg.lua:206]: Ok, my bad. The whole file is an example and does not exist anymore when the /etc/prosody volume is mounted.

**Review by madmatah — APPROVED** [review]

**Pull request merged**